frontend/dockerfile/instructions/commands_runmount.go

```diff
@@ -309,5 +305,9 @@ func parseMount(value string, expander SingleWordExpander) (*Mount, error) {
 		}
 	}
 
+	if m.CacheSharing != "" && m.Type != MountTypeCache {
+		return nil, errors.Errorf("invalid cache sharing set for %v mount", m.Type)
```

Comment:
We can remove the above `m.Type == MountTypeSecret` variant of this check entirely, and replace this with:

```go
		return nil, errors.Errorf("%s mount should not define sharing set", m.Type)
```

or similar.

Reply:
oh, I see I missed your comment; let me know if you want me to look at this